app/controllers/task_contributions_controller.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+class TaskContributionsController < ApplicationController
+  include TaskParameters
+
+  before_action :load_and_authorize_task, except: %i[show edit]
+  before_action :load_and_authorize_task_contribution, except: %i[create new]
+
+  def approve_changes
+    if @task.apply_contribution(@task_contribution)
+      TaskContributionMailer.approval_info(@task_contribution).deliver_later
+      redirect_to @task, notice: t('.success')
+    else
+      redirect_to [@task, @task_contribution], alert: t('.error')
+    end
+  end
+
+  def discard_changes
+    if @task_contribution.close
+      TaskContributionMailer.rejection_info(@task_contribution).deliver_later
```

Comment:
I would suggest not to send this email if the contribution was closed by the contributor.

Reply:
This issue has been fixed with 5fecfda92995c0f25a000a5c9ff56c81b5853749.